Library/Homebrew/utils/gems.rb

```diff
@@ -128,23 +128,33 @@ def install_bundler!
     )
   end
 
-  def install_bundler_gems!(only_warn_on_failure: false, setup_path: true)
+  def install_bundler_gems!(only_warn_on_failure: false, setup_path: true, groups: [])
     old_path = ENV["PATH"]
     old_gem_path = ENV["GEM_PATH"]
     old_gem_home = ENV["GEM_HOME"]
     old_bundle_gemfile = ENV["BUNDLE_GEMFILE"]
+    old_bundle_with = ENV["BUNDLE_WITH"]
 
     install_bundler!
 
+    require "settings"
+
+    groups |= (Homebrew::Settings.read(:gemgroups)&.split(";") || [])
+    groups.sort!
+
     ENV["BUNDLE_GEMFILE"] = File.join(ENV.fetch("HOMEBREW_LIBRARY"), "Homebrew", "Gemfile")
-    @bundle_installed ||= begin
+    ENV["BUNDLE_WITH"] = groups.join(" ")
+
+    if @bundle_installed_groups != groups
```

Comment:
Could this be an early `return`?

Reply:
There is a `setup_gem_environment!` below this.